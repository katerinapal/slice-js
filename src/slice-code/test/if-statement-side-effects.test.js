import {runAllCombosTests, comboOfBools} from './helpers/utils'

// test.only(
//   'ifWithAssignment(true)',
//   require('./helpers/utils').snapSlice(require.resolve('./fixtures/if-statement-side-effects'), ({ifWithAssignment}) => {
//     return [
//       ifWithAssignment(true),
//     ]
//   })
// )


runAllCombosTests({
  filename: require.resolve('./fixtures/if-statement-side-effects'),
  methods: [
    {methodName: 'ifWithAssignment', possibleArguments: comboOfBools(1)},
    {methodName: 'ifWithFunctionCall', possibleArguments: comboOfBools(1)},
  ],
})
