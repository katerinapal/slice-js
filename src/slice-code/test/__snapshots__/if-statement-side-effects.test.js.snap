// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ifWithAssignment(false) && ifWithAssignment(true) 1`] = `
"export {ifWithAssignment}

function ifWithAssignment(passIf) {
  let x
  if ((x = 'hi') && passIf) {
    return x + passIf
  } else {
    return !passIf + x
  }
}
"
`;

exports[`ifWithAssignment(false) 1`] = `
"export {ifWithAssignment}

function ifWithAssignment(passIf) {
  let x = 'hi'

  return !passIf + x
}
"
`;

exports[`ifWithAssignment(true) 1`] = `
"export {ifWithAssignment}

function ifWithAssignment(passIf) {
  let x = 'hi'

  return x + passIf
}
"
`;

exports[`ifWithFunctionCall(false) && ifWithFunctionCall(true) 1`] = `
"export {ifWithFunctionCall}

function ifWithFunctionCall(passIf) {
  let thing

  if (
    passIf &&
    (a => ((thing = a), passIf))('hey') &&
    (function(a) {
      return a
    })(passIf)
  ) {
    return thing + passIf
  } else {
    return !passIf
  }
}
"
`;

exports[`ifWithFunctionCall(false) 1`] = `
"export {ifWithFunctionCall}

function ifWithFunctionCall(passIf) {
  !(function(a) {
    return a
  })(passIf)

  return !passIf
}
"
`;

exports[`ifWithFunctionCall(true) 1`] = `
"export {ifWithFunctionCall}

function ifWithFunctionCall(passIf) {
  let thing

  passIf && (a => ((thing = a), passIf))('hey');
  (function(a) {
    return a
  })(passIf)

  return thing + passIf
}
"
`;
